### PR TITLE
REF: simplify sanitize_array

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

File: pandas/core/construction.py [hunk 1]
```diff
@@ -582,28 +602,27 @@ def sanitize_array(
...
-    elif isinstance(data, ABCExtensionArray):
-        # it is already ensured above this is not a PandasArray
-        subarr = data
-
-        if dtype is not None:
-            subarr = subarr.astype(dtype, copy=copy)
-        elif copy:
-            subarr = subarr.copy()
+    elif hasattr(data, "__array__"):
+        # e.g. dask array GH#38645
```
Comment (jbrockmendel) [inline pandas/core/construction.py:606]: @mroeschke one thought is here we could check for e.g. pyarrow array and avert a cast to ndarray.  most of the tests that get here do so with ChunkedArray and eventually cast back to ArrowArray

Comment (mroeschke) [reply pandas/core/construction.py:606]: So publicly this is the case when one calls e.g. `pd.Series(pa.chunked_array([[1, 2]]))`?

Comment (jbrockmendel) [reply pandas/core/construction.py:606]: right

Comment (mroeschke) [reply pandas/core/construction.py:606]: Okay cool. Will put up a PR in the next few days

**Pull request merged**